src/Grid/GridHeader.js

```diff
@@ -110,144 +93,111 @@ class GridHeader extends React.Component {
+      [this.state.activeFilter + name]: event.target.value
+    });
+  };
 
   handleBooleanDropDown = event => {
     this.setState({ [`${[this.state.activeFilter]}Value`]: event.target.value });
   };
 
-  DialogDropdown = props => {
-    const value = this.state[this.state.activeFilter] === undefined ? 'None' : this.state[this.state.activeFilter];
-    
-    if (this.state.columnType === 'string') {
-      return (<StringDropdown classes={props.classes} value={value} activeFilter={this.state.activeFilter} handleChange={this.handleChange.bind(this)}/>);
-    }
-    else if (this.state.columnType === 'numeric' || this.state.columnType === 'datetime' || this.state.columnType === 'date' || this.state.columnType === 'datetimeutc') {
-      return (<NumericDropdown classes={props.classes} value={value} activeFilter={this.state.activeFilter} handleChange={this.handleChange.bind(this)}/>);
-    }
-    else if(this.state.columnType === 'boolean'){
-      return (<BooleanDropdown classes={props.classes} value={value} activeFilter={this.state.activeFilter} handleChange={this.handleChange.bind(this)}/>);
-    }
-    else {
-      return (<div />);
-    }
-  }
-
   render() {
     const { dataSource, classes } = this.props;
 
     return (
       <TableHead>
         <Dialog open={this.state.open} onClose={this.handleClose} >
-          <this.DialogDropdown classes={classes} />
-          <this.DialogContent classes={classes} />
+          <DialogDropdown 
+            classes={classes}
+            value={this.state[this.state.activeFilter]}
+            columnType={this.state.columnType}
+            activeFilter={this.state.activeFilter}
+            handleChange={this.handleChange.bind(this)} 
+          /> 
+          <DialogContent
+            classes={classes}
+            columnType={this.state.columnType}
+            activeFilter={this.state.activeFilter}
+            operator={this.state[this.state.activeFilter]}
+            value={this.state[`${this.state.activeFilter}Value`]}
+            value2={this.state[`${this.state.activeFilter}Value2`]}
+            handleDatePicker={this.handleDatePicker.bind(this)}
+            handleBooleanDropDown={this.handleBooleanDropDown.bind(this)}
+            handleTextFieldChange={this.handleTextFieldChange.bind(this)}
+            handleApply={this.handleApply.bind(this)}
+            handleClear={this.handleClear.bind(this)} 
+          />          
         </Dialog>
         <TableRow>
           {dataSource.columns.map(column => {
             const render = column.Sortable ?
               (<Tooltip title='Sort' placement='bottom-start' enterDelay={300}>
-                <TableSortLabel
-                  onClick={() => this.sortHandler(column.Label, column.DataType)}
-                >
+                <TableSortLabel onClick={() => this.sortHandler(column.Name)} >
                   {column.Label}
+                  {column.SortOrder === 1 && column.SortDirection === 'Ascending' ? 
```

Comment:
The SortOrder should not be evaluated here... Please check the logic at tubular.